### ISSUE TITLE
doctor mcp_config reports OK when MCP auth token is wrong

### ISSUE THREAD

**Issue by joyshmitz** [Dicklesworthstone/mcp_agent_mail_rust#69]
`am doctor check` validates MCP config URL drift, but it appears not to validate bearer-token drift.

Relevant check path:
- `crates/mcp-agent-mail-cli/src/lib.rs:13925-13931`

Repair path already has access to the desired bearer token:
- `crates/mcp-agent-mail-cli/src/lib.rs:15014-15018`

Minimal repro:
```bash
home="$(mktemp -d)"
mkdir -p "$home/.config/mcp-agent-mail" "$home/.claude"
printf 'HTTP_BEARER_TOKEN=canonical-good-token\n' > "$home/.config/mcp-agent-mail/config.env"

cat > "$home/.claude/settings.json" <<'JSON'
{
  "mcpServers": {
    "mcp-agent-mail": {
      "type": "http",
      "url": "http://127.0.0.1:8765/mcp/",
      "headers": {
        "Authorization": "Bearer definitely-wrong-token"
      }
    }
  }
}
JSON

HOME="$home" \
XDG_CONFIG_HOME="$home/.config" \
am doctor check --json | jq '.checks[] | select(.check=="mcp_config")'
```

Observed:
- `mcp_config` reports `ok`
- the wrong bearer token is not surfaced

Expected:
- `doctor` should detect missing/mismatched auth headers without printing the secret.
